obelix/idefix/services/github/issue.go

```diff
@@ -0,0 +1,108 @@
+package github
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/Shopify/sarama"
+	redisv8 "github.com/go-redis/redis/v8"
+	"idefix/operator"
+	"idefix/producer"
+	"idefix/redis"
+)
+
+var ErrNoIssues = fmt.Errorf("no issues in the repository")
+
+type Issues struct {
+	Issues []Issue `json:"issues,omitempty"`
+}
+
+type Issue struct {
+	ID            int         `json:"id"`
+	NodeID        string      `json:"node_id"`
+	URL           string      `json:"url"`
+	RepositoryURL string      `json:"repository_url"`
+	PullRequest   interface{} `json:"pull_request,omitempty"`
+}
+
+func (i *Issues) Parse(data []byte) error {
+	err := json.Unmarshal(data, &i.Issues)
+
+	if err != nil {
+		return fmt.Errorf("failed to Unmarshal %v", err)
+	}
+	return nil
+}
+
+func (i *Issues) findLatestIssue() (*Issue, bool) {
+	for _, elem := range i.Issues {
+		if elem.PullRequest == nil {
+			return &elem, true
+		}
+	}
+	return nil, false
+}
+
+func (i *Issues) GetRedisState(rc *redis.Client, key string) (string, error) {
+	state, err := rc.GetKey(key)
+
+	if err == redisv8.Nil {
+		elem, ok := i.findLatestIssue()
+		if ok == false {
+			return "", ErrNoIssues
+		}
+
+		err = rc.SetKey(key, elem.URL)
+		if err != nil {
+			return "", fmt.Errorf("failed to set value in redis: %v", err)
+		}
+		fmt.Println("Create value in redis: ", elem.URL)
+
+	} else if err != nil {
+		return "", fmt.Errorf("failed to get value from redis: %v", err)
+	}
+	return state, nil
+}
+
+func (i *Issues) LookForChange(op *operator.IdefixOperator, key, old string) error {
+	newer, ok := i.findLatestIssue()
+	if ok == false {
+		return ErrNoIssues
+	}
+
+	if old != newer.URL {
+		fmt.Println("That news :): ", newer.URL)
```

Comment:
Remove this debug print.

Reply:
6572d568177a397940d5454fb1d43e8c72a00a97